bblfsh/compat.py

```diff
@@ -0,0 +1,228 @@
+    Mode, VersionResponse, DESCRIPTOR
+)
+from bblfsh.pyuast import uast, iterator as native_iterator
+from bblfsh.tree_order import TreeOrder
+
+if "BBLFSH_COMPAT_SHUTUP" not in os.environ:
+    print("Warning: using deprecated bblfsh v1 compatibility layer.",
+          file=sys.stderr)
+
+
+class WrongTypeException(Exception):
+    pass
+
+
+class CompatParseResponse:
+    def __init__(self, ctx: ResultContext, filename: str = "") -> None:
+        self._res_context = ctx
+        self._filename = filename
+
+    @property
+    def uast(self) -> Node:
+        return self._res_context.uast
+
+    @property
+    def ast(self) -> Node:
+        return self._res_context.ast
+
+    @property
+    def ctx(self) -> ResultContext:
+        return self._res_context
+
+    @property
+    def elapsed(self) -> int:
+        # FIXME(juanjux): check if the can get this or measure ourselves
+        return -1
+
+    @property
+    def language(self) -> str:
+        return self._res_context.language
+
+    @property
+    def filename(self) -> str:
+        return self._filename
+
+    @property
+    def DESCRIPTOR(self) -> Any:
+        return self._res_context.ctx.DESCRIPTOR
+
+    @property
+    def errors(selfs) -> List:
+        # ParseResponse would have raised an exception on errors
+        return []
+
+
+class CompatBblfshClient:
+    def __init__(self, endpoint: Union[str, grpc.Channel]) -> None:
+        self._bblfsh_cli = bcli.BblfshClient(endpoint)
+
+        self._channel = self._bblfsh_cli._channel
+        self._stub_v1 = self._bblfsh_cli._stub_v1
+        self._stub_v2 = self._bblfsh_cli._stub_v2
+
+    def _parse(self, filename: str, language: str = None, contents: str = None,
+               timeout: float = None,
+               mode: ModeType = Mode.Value('ANNOTATED')) -> CompatParseResponse:
+
+        if timeout is not None:
+            timeout = int(timeout)
+
+        res = self._bblfsh_cli.parse(filename, language, contents,
+                                     mode=mode, timeout=timeout)
+        return CompatParseResponse(res, filename)
+
+    def parse(self, filename: str, language: str = None, contents: str = None,
+              timeout: float = None) -> CompatParseResponse:
+
+        return self._parse(filename, language, contents, timeout,
+                           Mode.Value('ANNOTATED'))
+
+    def native_parse(self, filename: str, language: str = None,
+                     contents: str = None,
+                     timeout: float = None) -> CompatParseResponse:
+
+        return self._parse(filename, language, contents, timeout,
+                           Mode.Value('NATIVE'))
+
+    def supported_languages(self) -> List[str]:
+        return self._bblfsh_cli.supported_languages()
+
+    def version(self) -> VersionResponse:
+        return self._bblfsh_cli.version()
+
+    def close(self) -> None:
+        return self._bblfsh_cli.close()
+
+
+class CompatNodeIterator:
+    def __init__(
```

Comment:
[nit] Even with an 80-column limit, this fits on one line. (Well, okay it's exactly 80 so the line break rolls over—calling the iterator `it` would fix that ;)

Reply:
Out [current conventions](https://github.com/src-d/guide/blob/master/engineering/conventions/python.md) specify a 99 line length so I'll merge all.